unity-renderer/Assets/Rendering/Culling/CullingObjectsTracker.cs

```diff
@@ -1,144 +1,198 @@
-    /// to optimize a bigger bottleneck.
-    /// </summary>
+    /// <summary> This class is used for tracking all the renderers, skinnedMeshRenderers and Animations of the world. </summary>
     public class CullingObjectsTracker : ICullingObjectsTracker
     {
-        private ICollection<Renderer> renderers;
-        private ICollection<SkinnedMeshRenderer> skinnedRenderers;
-        private Animation[] animations;
+        private readonly List<Rendereable> rendereables = new List<Rendereable>();
+        private ICollection<Renderer> usingRenderers = new List<Renderer>();
+        private ICollection<SkinnedMeshRenderer> detectedSkinnedRenderers;
+        private ICollection<SkinnedMeshRenderer> usingSkinnedRenderers = new List<SkinnedMeshRenderer>();
+        private Animation[] usingAnimations;
 
         private int ignoredLayersMask;
         private bool dirty = true;
-
+        
 
         public CullingObjectsTracker()
         {
             PoolManager.i.OnGet -= MarkDirty;
             PoolManager.i.OnGet += MarkDirty;
+
+            Rendereable.OnRendereableAdded -= OnRendereableAdded;
+            Rendereable.OnRendereableAdded += OnRendereableAdded;
         }
 
-        /// <summary>
-        /// If the dirty flag is true, this coroutine will re-populate all the tracked objects.
-        /// </summary>
+        /// <summary> Adds a rendereable to be tracked and calculating in the culling process. </summary>
+        public void OnRendereableAdded(object sender, Rendereable rendereable) => rendereables.Add(rendereable);
+
+        /// <summary> Sets a layer mask to be ignored, only one layer mask can be ingored at a given time. </summary>
+        public void SetIgnoredLayersMask(int ignoredLayersMask) { this.ignoredLayersMask = ignoredLayersMask; }
+
+        /// <summary> This will re-populate all the tracked objects synchronously </summary>
+        public void ForcePopulateRenderersList()
+        {
+            detectedSkinnedRenderers = Object.FindObjectsOfType<SkinnedMeshRenderer>();
+
+            ForceCalculateRenderers();
+            ForceCalculateSkinnedRenderers();
+            usingAnimations = Object.FindObjectsOfType<Animation>();
+
+            dirty = false;
+        }
+
+        /// <summary> If the dirty flag is true, this coroutine will re-populate all the tracked objects. </summary>
         public IEnumerator PopulateRenderersList()
         {
             if (!dirty)
                 yield break;
 
-            List<Renderer> renderersList = new List<Renderer>();
-            List<SkinnedMeshRenderer> skinnedRenderersList = new List<SkinnedMeshRenderer>();
-            Renderer[] allRenderers = Object.FindObjectsOfType<Renderer>();
-
-            // TODO new task, avoid FindGameObjectsOfType and use:
-            // SceneManager.GetActiveScene().GetRootGameObjects(), calculate from there.
+            detectedSkinnedRenderers = Object.FindObjectsOfType<SkinnedMeshRenderer>();
```

Comment:
I am sorry, I didn't get the changes: so instead of `Object.FindObjectsOfType<Renderer>();` you call `Object.FindObjectsOfType<SkinnedMeshRenderer>()` and `Object.FindObjectsOfType<Animation>()`?
What's the difference? Performance-wise?

Reply:
Finding `Object.FindObjectsOfType<SkinnedMeshRenderer>()` costs 10^-5 seconds and `Object.FindObjectsOfType<Animation>()` a bit less than that, the spike was `Object.FindObjectsOfType<Renderer>();` and that is the one I aimed to solve in this PR

Reply:
could we have any issue about deactivated SkinnedMeshRenderer/Animation/Renderer at the given moment of the query?

Reply:
There has been no changes for SkinnedMeshRenderers and Animations in that regard, so no issues should appear (That were non existent before)